src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -1649,5 +1675,179 @@ private bool HasErrorsSlow()
 
             return clone;
         }
+
+        /// <summary>
+        /// Gets the index of the given <paramref name="targetNode"/> in the current SyntaxNode.
+        /// </summary>
+        /// <returns>
+        /// <c>-1</c> if <paramref name="targetNode"/> isn't a direct child.
+        /// </returns>
+        internal int GetIndexOfChild(SyntaxNode targetNode)
+        {
+            // Since we bypass the enumeration of a ChildSyntaxList,
+            // maintaining a binary search for as long as we can is best,
```

Comment:
can you explain how this is a binary search?

Reply:
this comment should probably be moved to the `TryGetSlotOfListChild` method that it's talking about,
where most of the searching happens (its logic was moved into a new method for readability reasons etc, but it being inlined is perf-critical since it's a hot-path with compound effects on analysis perf)

Reply:
ok.  will wait for the update.